expression/src/main/java/io/smallrye/common/expression/Expression.java

```diff
@@ -485,15 +507,30 @@ private static Node parseString(Itr itr, final boolean allowExpr, final boolean
                             }
                             //throw Assert.unreachableCode();
                         }
+                        case '\\': {
+                            if (flags.contains(Flag.NO_$$)) {
+                                if (itr.hasNext() && itr.peekNext() == '$') {
+                                    list.add(new LiteralNode(itr.getStr(), start, idx));
+                                    start = itr.getNextIdx();
+                                    itr.next();
+                                    continue;
+                                }
+                            }
+                        }
```

Comment:
This shouldn't be needed anymore AFAICT.

Reply:
The goal here was to support the MP escape sequence of \$, without turning on ESCAPES, which breaks commas separated values escapes.